src/main/java/fi/aalto/cs/apluscourses/presentation/base/SearchableNode.java

```diff
@@ -0,0 +1,10 @@
+package fi.aalto.cs.apluscourses.presentation.base;
+
+import org.jetbrains.annotations.NotNull;
+
+public interface SearchableNode {
+  @NotNull
+  default String getSearchableString() {
+    return "";
+  }
+}
```

Comment:
Sorry I was a bit unclear, but I mainly suggested this as a workaround before I discovered the `Treeview.getViewModel` method, so I would say this isn't really needed anymore

Reply:
I like this in the way that it keeps the lambda in TreeSpeedSearch's constructor short, and moves the "get searchable string" logic from View to the ViewModel

Reply:
Alright :+1: Just one nit, can we then remove `Node` from the name, since this isn't limited to tree nodes